src/pudl_archiver/archivers/classes.py

```diff
@@ -245,7 +249,86 @@ def _check_missing_files(
             description="Check for files from previous version of archive that would be deleted by the new version",
             success=len(missing_files) == 0,
             notes=notes,
-            required_for_run_success=self.check_missing_files,
+            required_for_run_success=self.fail_on_missing_files,
+        )
+
+    def _check_file_size(
+        self,
+        baseline_datapackage: DataPackage | None,
+        new_datapackage: DataPackage,
+    ) -> validate.DatasetSpecificValidation:
+        """Check if any one file's size has changed by |>allowed_file_rel_diff|."""
+        baseline_resources = set()
```

Comment:
These two start out as `set`s but are treated as `dict`s later on, which has some surprising behavior:

```
>>> ss = set()
>>> ss.update({1: "a"})
>>> ss
{1}
```

We should probably just initialize them as dicts here.

Reply:
Ah I did this because `ruff` was screaming at me but I'll update it, that seems undesirable.